### PR TITLE
Simplify numeric.h

### PR DESCRIPTION
**Pull request by asi1024**

### CODE REVIEW

File: chainerx_cc/chainerx/cuda/numeric.cuh [hunk 1]
```diff
@@ -22,89 +22,28 @@ __device__ inline bool IsInf(cuda::Float16 value) { return value.IsInf(); }
...
+DECLARE_CUDA_FLOAT16_FALLBACK_UNARY(Tanh, std::tanh(x))
+DECLARE_CUDA_FLOAT16_FALLBACK_UNARY(Sin, std::sin(x))
+DECLARE_CUDA_FLOAT16_FALLBACK_UNARY(Cos, std::cos(x))
+DECLARE_CUDA_FLOAT16_FALLBACK_UNARY(Tan, std::tan(x))
+DECLARE_CUDA_FLOAT16_FALLBACK_UNARY(Arcsin, std::asin(x))
+DECLARE_CUDA_FLOAT16_FALLBACK_UNARY(Arccos, std::acos(x))
+DECLARE_CUDA_FLOAT16_FALLBACK_UNARY(Arctan, std::atan(x))
+DECLARE_CUDA_FLOAT16_FALLBACK_UNARY(Exp, std::exp(x))
+DECLARE_CUDA_FLOAT16_FALLBACK_UNARY(Log, std::log(x))
+DECLARE_CUDA_FLOAT16_FALLBACK_UNARY(Sqrt, std::sqrt(x))
```
Comment (hvy) [inline chainerx_cc/chainerx/cuda/numeric.cuh]: I think prefixing with `DEFINE` is more correct. What do you say?

Comment (hvy) [inline chainerx_cc/chainerx/cuda/numeric.cuh]: Looks simpler and sufficient but are these cases where this isn't appropriate?
```suggestion
DECLARE_CUDA_FLOAT16_FALLBACK_UNARY(Sqrt, std::sqrt)
```

**Comment by hvy** [comment]
Change the label from "code-fix" since this PR actually fixes Square.

File: chainerx_cc/chainerx/cuda/numeric.cuh (hunk 1 above)
Comment (asi1024) [reply chainerx_cc/chainerx/cuda/numeric.cuh]: Sometimes `x` may appear twice in the expression in the future.

Comment (hvy) [reply chainerx_cc/chainerx/cuda/numeric.cuh]: I see, could you for the record also mention them here?

Comment (asi1024) [reply chainerx_cc/chainerx/cuda/numeric.cuh]: Fixed.

Comment (asi1024) [reply chainerx_cc/chainerx/cuda/numeric.cuh]: Fixed to remove `x`. In the above case, it is OK to define a unary function ahead and pass that as the second argument to the macro.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Jenkins failure seem unrelated. Waiting for Travis.

**Comment by hvy** [comment]
LGTM!

**Comment by hvy** [comment]
Removing `to-be-backported` as we decided not to backport non-critical PRs.

**Pull request merged**